scripts/create_or_validate_md5s.py

```diff
@@ -0,0 +1,159 @@
+import os
+
+import click
+from google.cloud import storage
+
+from cpg_utils.hail_batch import config_retrieve, copy_common_env, get_batch
+
+
+def get_blobs_in_directory(gs_dir: str, billing_project: str, storage_client: storage.Client):
+    """Get all blobs in a directory as a set of full gs:// URIs"""
+    bucket_name, *components = gs_dir[5:].split('/')
+    bucket = storage_client.bucket(bucket_name, user_project=billing_project)
+    blobs = bucket.list_blobs(prefix='/'.join(components))
+    return {f'gs://{bucket_name}/{blob.name}' for blob in blobs}
+
+
+def create_md5(job, filepath: str, billing_project: str, driver_image: str):
+    """
+    Streams the file with gsutil and calculates the md5 checksum,
+    then uploads the checksum to the same path as filename.md5.
+    """
+    copy_common_env(job)
+    job.image(driver_image)
+    md5_filepath = f'{filepath}.md5'
+    job.command(
+        f"""\
+    set -euxo pipefail
+    gcloud -q auth activate-service-account --key-file=$GOOGLE_APPLICATION_CREDENTIALS
+    gsutil -u {billing_project} cat {filepath} | md5sum | cut -d " " -f1  > /tmp/uploaded.md5
+    gsutil -u {billing_project} cp /tmp/uploaded.md5 {md5_filepath}
+    """
+    )
+
+    return job
+
+
+def validate_md5(job, filepath: str, billing_project: str, driver_image: str):
+    """
+    Streams the file with gsutil and calculates the md5 checksum,
+    then compares it to the stored checksum in the .md5 file.
+    """
+    copy_common_env(job)
+    job.image(driver_image)
+    md5_filepath = f'{filepath}.md5'
+    job.command(
+        f"""\
+    set -euxo pipefail
+    gcloud -q auth activate-service-account --key-file=$GOOGLE_APPLICATION_CREDENTIALS
+    calculated_md5=$(gsutil -u {billing_project} cat {filepath} | md5sum | cut -d " " -f1)
```

Comment:
Nitpicky / petty, but good to use shlex.quote these params, ie: if a user provides the billing_project `"help); gsutil rm -rf gs://bucket; calculated_md5=$(gsutil -u seqr"`, this could cause a lot of problems.